specifyweb/specify/tree_ranks.py

```diff
@@ -278,3 +279,41 @@ def set_rank_id(new_rank):
 
     # Set the new rank id
     new_rank.rankid = new_rank_id
+
+class RankOperation(Enum):
+    CREATED = 'created'
+    DELETED = 'deleted'
+    UPDATED = 'updated'
+
+def verify_rank_parent_chain_integrity(rank, rank_operation: RankOperation):
```

Comment:
this looks like a good check to guard against https://github.com/specify/specify7/pull/4257#discussion_r1544051679

but my question still remains:
- is such check necessary? i.e since rank is referring to it's parent by ID, does database enforce this integrity already, or do we have to do that manually using the verify_rank_parent_chain_integrity function. i.e could you use our back-end API as it stands right now (without verify_rank_parent_chain_integrity) to intentionally break the tree def integrity? if not, then this check is needless
- if this check is needed, in what cases is it needed? does any tree rank update cause issues? if so, we must handle the rank rename logic. or is it only needed on database migrations?